torch_xla/csrc/ops/dynamic_ir.h

```diff
@@ -53,6 +53,21 @@ class SizeNode : public XlaNode, public torch::lazy::DimensionNode {
   mutable int64_t runtime_size_;
 };
 
+class SizeEq : public XlaNode, public torch::lazy::DimensionNode {
+ public:
+  SizeEq(torch::lazy::Value a, torch::lazy::Value b);
+  int64_t getDynamicValue() const override;
+  int64_t getStaticValue() {
+    TORCH_CHECK(false, "Comparison operators should be using getDynamicValue");
```

Comment:
Intuitively, I thought 2 nodes are equal when both static value and dynamic value are equal respectively. I wonder why we are only comparing dynamic values here.

Reply:
That's because conditional operations work off of "true" dynamic tensor values. Look under conditional operations in the design doc for more detail.